internal/engine/actions/actions.go

```diff
@@ -138,22 +138,82 @@ func (rae *RuleActionsEngine) processAction(
 	params engif.ActionsParams,
 	metadata *json.RawMessage,
 ) (json.RawMessage, error) {
+	logger := zerolog.Ctx(ctx).Debug().
+		Str("action", string(actionType)).
+		Str("cmd", string(cmd))
+
+	// Exit early in case we shouldn't do anything, i.e., cmd != ActionCmdOn or cmd != ActionCmdOff
+	if cmd == engif.ActionCmdDoNothing {
+		logger.Msg("skipping action")
+		switch actionType {
+		case remediate.ActionType:
+			// Return the previous remediation status.
+			err := enginerr.RemediationStatusAsError(params.GetEvalStatusFromDb().RemStatus.RemediationStatusTypes)
+			return *(getMeta(params.GetEvalStatusFromDb().RemMetadata)), err
+		case alert.ActionType:
+			// Return the previous alert status.
+			err := enginerr.AlertStatusAsError(params.GetEvalStatusFromDb().AlertStatus.AlertStatusTypes)
+			return *(getMeta(params.GetEvalStatusFromDb().AlertMetadata)), err
+		}
```

Comment:
The above might read more easily if you extract the `params.GetEvalStatusFromDb()` that's used 4 times into an 'evalStatus' temporary.

```suggestion
		logger.Msg("skipping action")
		evalStatus := params.GetEvalStatusFromDb()
		switch actionType {
		case remediate.ActionType:
			// Return the previous remediation status.
			err := enginerr.RemediationStatusAsError(evalStatus.RemStatus.RemediationStatusTypes)
			return *(getMeta(evalStatus.RemMetadata)), err
		case alert.ActionType:
			// Return the previous alert status.
			err := enginerr.AlertStatusAsError(evalStatus.AlertStatus.AlertStatusTypes)
			return *(getMeta(evalStatus.AlertMetadata)), err
		}
```